jpa/subsystem/src/main/java/org/jboss/as/jpa/messages/JpaLogger.java

```diff
@@ -753,8 +753,8 @@ public interface JpaLogger extends BasicLogger {
 
     // id = 72, value = "Could not obtain TransactionListenerRegistry from transaction manager")
 
-    @Message(id = 73, value = "Transformation of class %s failed")
-    IllegalStateException invalidClassFormat(@Cause Exception cause, String className);
+    @Message(id = 73, value = "Bytecode rewrite (transformation) of class %s failed")
+    IllegalStateException invalidClassFormat(String className);
```

Comment:
This should return a String, and the caller can then use the String in an ERROR log message and to create the ISE.

By returning the ISE you end up logging its toString(), so something like

15:24:24,703 ERROR  [org.jboss.as.jpa] (MSC service thread 1-7) java.lang.IllegalStateException: WFLYJPA0073: Bytecode rewrite (transformation) of class com.foo.bar failed

That 'java.lang.IllegalStateException: ' in the middle is confusing noise.

Reply:
The server log output looks better, thanks Brian!  Server log:
> 10:03:50,829 ERROR [org.jboss.as.jpa] (MSC service thread 1-1) WFLYJPA0073: Bytecode rewrite (transformation) of class org/jboss/as/quickstart/hibernate/model/Member failed: jakarta.persistence.spi.TransformerException: Error performing enhancement of org/jboss/as/quickstart/hibernate/model/Member
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.jpa.internal.enhance.EnhancingClassTransformerImpl.transform(EnhancingClassTransformerImpl.java:54)
        at org.jboss.as.jpa@30.0.0.Beta1-SNAPSHOT//org.jboss.as.jpa.processor.JPADelegatingClassFileTransformer$1.run(JPADelegatingClassFileTransformer.java:74)
        at org.jboss.as.jpa@30.0.0.Beta1-SNAPSHOT//org.jboss.as.jpa.processor.JPADelegatingClassFileTransformer$1.run(JPADelegatingClassFileTransformer.java:58)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.manager.WildFlySecurityManager.doChecked(WildFlySecurityManager.java:664)
        at org.jboss.as.jpa@30.0.0.Beta1-SNAPSHOT//org.jboss.as.jpa.processor.JPADelegatingClassFileTransformer.transform(JPADelegatingClassFileTransformer.java:91)
        at org.jboss.as.server@22.0.0.Beta1//org.jboss.as.server.deployment.module.DelegatingClassTransformer.transform(DelegatingClassTransformer.java:59)
        at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:466)
        at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:276)
        at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:79)
        at org.jboss.modules.Module.loadModuleClass(Module.java:765)
        at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
        at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
        at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
        at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
        at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
        at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
        at org.jboss.as.server@22.0.0.Beta1//org.jboss.as.server.deployment.reflect.ClassReflectionIndex.<init>(ClassReflectionIndex.java:80)
        at org.jboss.as.server@22.0.0.Beta1//org.jboss.as.server.deployment.reflect.DeploymentReflectionIndex.getClassIndex(DeploymentReflectionIndex.java:70)
        at org.jboss.as.ee@30.0.0.Beta1-SNAPSHOT//org.jboss.as.ee.metadata.MethodAnnotationAggregator.runtimeAnnotationInformation(MethodAnnotationAggregator.java:57)
        at org.jboss.as.ee@30.0.0.Beta1-SNAPSHOT//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.handleAnnotations(InterceptorAnnotationProcessor.java:106)
        at org.jboss.as.ee@30.0.0.Beta1-SNAPSHOT//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.processComponentConfig(InterceptorAnnotationProcessor.java:85)
        at org.jboss.as.ee@30.0.0.Beta1-SNAPSHOT//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.deploy(InterceptorAnnotationProcessor.java:77)
        at org.jboss.as.server@22.0.0.Beta1//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:188)
        at org.jboss.msc@1.5.1.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1617)
        at org.jboss.msc@1.5.1.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1580)
        at org.jboss.msc@1.5.1.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1438)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.hibernate.bytecode.enhance.VersionMismatchException: Mismatch between Hibernate version used for bytecode enhancement (6.2.0.Final) and runtime (6.3.0-SNAPSHOT) for `org.jboss.as.quickstart.hibernate.model.Member`
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.bytecode.enhance.internal.bytebuddy.EnhancerImpl.verifyVersions(EnhancerImpl.java:407)
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.bytecode.enhance.internal.bytebuddy.EnhancerImpl.doEnhance(EnhancerImpl.java:183)
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.bytecode.enhance.internal.bytebuddy.EnhancerImpl.lambda$enhance$0(EnhancerImpl.java:151)
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.bytecode.internal.bytebuddy.ByteBuddyState.rewrite(ByteBuddyState.java:163)
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.bytecode.enhance.internal.bytebuddy.EnhancerImpl.enhance(EnhancerImpl.java:151)
        at org.hibernate@6.3.0-SNAPSHOT//org.hibernate.jpa.internal.enhance.EnhancingClassTransformerImpl.transform(EnhancingClassTransformerImpl.java:51)
        ... 32 more


Just for fyi, the Maven command that I am executing to deploy the quickstarts/hibernate gets:
> [ERROR] Failed to execute goal org.wildfly.plugins:wildfly-maven-plugin:4.0.0.Final:deploy (default-cli) on project hibernate: Failed to execute goal deploy: {"WFLYCTL0062: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-1" => {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"hibernate.war\".POST_MODULE" => "WFLYSRV0153: Failed to process phase POST_MODULE of deployment \"hibernate.war\"
[ERROR]     Caused by: java.lang.RuntimeException: WFLYSRV0177: Error getting reflective information for class org.jboss.as.quickstart.hibernate.service.MemberRegistration with ClassLoader ModuleClassLoader for Module \"deployment.hibernate.war\" from Service Module Loader
[ERROR]     Caused by: java.lang.ClassFormatError: Failed to link org/jboss/as/quickstart/hibernate/model/Member (Module \"deployment.hibernate.war\" from Service Module Loader): WFLYJPA0073: Bytecode rewrite (transformation) of class org/jboss/as/quickstart/hibernate/model/Member failed"}}}}